File: rs/minsk/src/analysis/text_span.rs
#[derive(Clone, Copy)]
pub struct TextSpan {
    start: usize,
    length: usize,
}

impl TextSpan {
    pub fn new(start: usize, length: usize) -> Self {
        Self { start, length }
    }

    pub fn from_bounds(start: usize, end: usize) -> Self {
        Self {
            start,
            length: end - start,
        }
    }

    pub fn start(&self) -> usize {
        self.start
    }

    pub fn length(&self) -> usize {
        self.length
    }

    pub fn end(&self) -> usize {
        self.start + self.length
    }
}

File: nim/src/minsk/codeAnalysis/textSpan.nim
type
  TextSpan* = object
    start*: int
    length*: int

proc newTextSpan*(start, length: int): TextSpan =
  result.start = start
  result.length = length

proc fromBounds*(start, stop: int): TextSpan =
  newTextSpan(start, stop - start)

proc getEnd*(span: TextSpan): int =
  span.start + span.length
